### PR TITLE
Add pagination to list of devices in a group

### PR DESCRIPTION
**Pull request by CalumMcCall**
- PRO-2281

### CODE REVIEW

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala [hunk 1]
```diff
@@ -100,7 +99,7 @@ class GroupsResource(namespaceExtractor: Directive1[AuthedNamespaceScope], devic
       (scope.get & pathEnd) {
         listGroups(ns)
       } ~
-      (scope.get & extractGroupId & path("devices") ) { groupId =>
+      (scope.get & extractGroupId & path("devices") & pathEnd ) { groupId =>
```
Comment (Danten) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: Do you really need this `pathEnd`?

File: device-registry/src/main/scala/org/genivi/sota/device_registry/db/GroupMemberRepository.scala [hunk 2]
```diff
@@ -48,10 +52,28 @@ object GroupMemberRepository {
 
   def listDevicesInGroup(groupId: Uuid)(implicit ec: ExecutionContext): DBIO[Seq[Uuid]] =
     groupMembers
-      .filter(_.groupId === groupId)
-      .map(_.deviceUuid)
+      .filter(_.groupId === groupId).join(DeviceRepository.devices)
+      .on(_.deviceUuid === _.uuid)
+      .map(_._1.deviceUuid)
```
Comment (Danten) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/db/GroupMemberRepository.scala]: Should this be here?

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala (hunk 1 above)
Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: It is used in other endpoints that use parameters, but it seems like it isn't needed.

**Comment by CalumMcCall** [comment]
Yeah, that's something we can look at when we refactor the returned list to include offset/limit and total number of values so the client doesn't need to keep track of that info.

**Pull request merged**